libsel4vm/src/arch/arm/vgic/vgic.c

```diff
@@ -192,6 +191,14 @@ struct lr_of {
     bool full;
 };
 
+static inline size_t LR_OF_NEXT(size_t idx)
+{
+    struct lr_of *dummy_lr_of; /* need this to use ARRAY_SIZE() */
+    const size_t max = ARRAY_SIZE(dummy_lr_of->irqs) - 1;
+    assert(idx < max); /* the index must be sane when calling this */
```

Comment:
Maybe I'm confused but this should be `idx <= max` right?

Reply:
Thanks, yes, shame on me. This was `cnt` first, but then I made this `max`.

Reply:
Easy mistake to make :)